kong/plugins/ai-proxy/handler.lua

```diff
@@ -82,33 +82,72 @@ end
 
 
 function _M:body_filter(conf)
-  if kong.ctx.shared.skip_response_transformer then
-    return
-  end
-
-  if (kong.response.get_status() ~= 200) and (not kong.ctx.plugin.ai_parser_error) then
+  -- if body_filter is called twice, then return
+  if kong.ctx.plugin.body_called then
     return
   end
 
-  -- (kong.response.get_status() == 200) or (kong.ctx.plugin.ai_parser_error)
-
-  -- all errors MUST be checked and returned in header_filter
-  -- we should receive a replacement response body from the same thread
+  if kong.ctx.shared.skip_response_transformer then
+    local response_body, status
+    if kong.ctx.shared.parsed_response then
+      response_body = kong.ctx.shared.parsed_response
+    elseif kong.response.get_status() == 200 then
+      status, response_body = pcall( function()
```

Comment:
Any specific reason you wrap `kong.service.get_raw_body` in a pcall?

Reply:
@AntoineJac  ?

Reply:
Fix in last commit